Jaffenheimer/p7-internet
# Added user integration between frontend and backend (using endpoints)

Features:
- The user can now use forgot password, reset password and confirm email.
- Added modal for settings, (in dropdown) allowing the user to confirm their email and change their password.
- Added frontend and backend fixes to support forgot password

Before using:
Open docker
start restapi with dotnet run in the restapi folder or run the .NET restapi project
create a user using the frontend

(test for possible errors using the APIs)

User guide:
- **Confirm the email while logged in**
-- Requirement: Stay logged in on your user
-- goto settings modal by clicking the profile picture > "indstillinger" in the dropdown
-- Now press "send kode" (this will be sent automatically when creating a user later, limited to 100 mails per day)
-- Now type in the verification code and click "verificer"
-- The user has now verified their user, and is now able to use the forgot password feature

- **Use the forgot password (logged out)**
-- Requirement: the user has confirmed their email
-- Logout of your user
-- Goto the login box > glemt kodeord
-- Send verification code to your email by typing the assigned email to the created user and press "send" (check junk folder)
-- Type a new password and the verification code below and press "gendan". 
-- The user has now changed their password to the new password

- **Reset the password while logged in**
-- Requirement: Stay logged in on your user
-- goto settings modal by clicking the profile picture > "indstillinger" in the dropdown
-- Now press the link on "skift kodeord her"
-- Fill out the old password and the new password and repeated new password
-- Press the "skift kodeord" to change the password.
-- The user has now changed their password successfully.
